src/deploy.py

```diff
@@ -28,6 +28,7 @@ def impose_ordering(branch_list):
 @echo_output
 def deploy(pname, cluster=None, branch='master'):
     pdata = project.project_data(pname)
+    assert pdata['repo'], ('The `deploy` task needs a `repo` to be specified in the project %s configuration' % pname)
```

Comment:
this case should be handled in the `@requires_branch_deployable_project` - the logic should proceed with the assumption that a project with a valid repo has been supplied. if that decorator is letting these projects past, then it's broken.